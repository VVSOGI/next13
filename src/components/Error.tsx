'use client'
import React from 'react'

interface Props {
    message: string
}

export function Error({ message }: Props) {
    return (
        <div>
            <h2>{message}</h2>
            <button>Try again</button>
        </div>
    )
}
