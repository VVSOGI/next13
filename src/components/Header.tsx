import React from 'react'

export function Header() {
    return (
        <div
            style={{
                padding: '3rem 3rem',
                borderBottom: '1px solid #eaeaea',
                flex: 'none'
            }}
        >
            Header
        </div>
    )
}
