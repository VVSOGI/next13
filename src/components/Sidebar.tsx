import React from 'react'

interface Props {
    style?: React.CSSProperties
}

export function Sidebar({ style }: Props) {
    return (
        <div
            style={{
                ...style,
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
                height: '100%',
                borderRight: '1px solid #eaeaea'
            }}
        >
            Sidebar
        </div>
    )
}
