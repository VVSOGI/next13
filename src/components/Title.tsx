import React, { useState } from 'react'

interface Props {
    title: string
    changeTitle: (title: string) => void
}

export function Title({ title, changeTitle }: Props) {
    return (
        <div style={{ display: 'flex', flexDirection: 'column' }}>
            {title}
            <input
                style={{ width: '400px', padding: '1rem', marginTop: '1rem' }}
                placeholder="Enter your home title"
                type="text"
                onChange={(e) => changeTitle(e.target.value)}
            />
        </div>
    )
}
