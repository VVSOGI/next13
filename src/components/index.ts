'use client'

export * from './Header'
export * from './Title'
export * from './projects'
export * from './Error'
