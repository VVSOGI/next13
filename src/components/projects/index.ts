export * from './ProjectsContainer'
export * from './ProjectContents'
