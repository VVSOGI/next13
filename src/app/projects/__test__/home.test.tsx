import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { ProjectsContainer } from '@/components'

describe('Projects', () => {
    const customFetch = jest.fn()

    const renderProjects = async (mockProjects: Project[]) => {
        customFetch.mockResolvedValueOnce(mockProjects)
        const projects = await customFetch()

        return render(<ProjectsContainer projects={projects} />)
    }

    test('renders projects', async () => {
        const mockProjects: Project[] = [
            { id: 1, title: 'Project 1' },
            { id: 2, title: 'Project 2' }
        ]

        const { container } = await renderProjects(mockProjects)
        expect(container).toMatchSnapshot()
    })

    test('if projects is empty', async () => {
        const mockProjects: Project[] = []

        const { container } = await renderProjects(mockProjects)
        expect(container).toMatchSnapshot()
    })
})
