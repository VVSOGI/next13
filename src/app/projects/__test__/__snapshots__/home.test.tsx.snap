// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects if projects api is error 1`] = `
<div>
  <div>
    <h2>
      API is error
    </h2>
    <button>
      Try again
    </button>
  </div>
</div>
`;

exports[`Projects if projects is empty 1`] = `
<div>
  <section
    style="display: flex; align-items: center; background-color: rgb(255, 255, 255); color: rgb(2, 7, 37);"
  />
</div>
`;

exports[`Projects renders projects 1`] = `
<div>
  <section
    style="display: flex; align-items: center; background-color: rgb(255, 255, 255); color: rgb(2, 7, 37);"
  >
    <div>
      Project 1
    </div>
    <div>
      Project 2
    </div>
  </section>
</div>
`;
